src/adapter/src/coord/message_handler.rs

```diff
@@ -430,6 +433,47 @@ impl Coordinator {
             .await;
     }
 
+    #[tracing::instrument(level = "debug", skip(self, tx, session))]
+    async fn message_create_connection_validation_ready(
+        &mut self,
+        CreateConnectionValidationReady {
+            session,
+            tx,
+            result,
+            params,
+            original_stmt,
+            depends_on,
+            otel_ctx,
+        }: CreateConnectionValidationReady,
+    ) {
+        otel_ctx.attach_as_parent();
+
+        // Ensure that all dependencies still exist after validation, as a
+        // `DROP SECRET` may have sneaked in. If any have gone missing, we
+        // revalidate the original statement. This will either produce a nice
+        // "unknown secret" error, or pick up a new connector that has
+        // replaced the dropped connector.
+        //
+        // WARNING: If we support `ALTER SECRET`, we'll need to also check
+        // for connectors that were altered while we were purifying.
+        if !depends_on
+            .iter()
+            .all(|id| self.catalog().try_get_entry(id).is_some())
+        {
+            self.handle_execute_inner(original_stmt, params, session, tx)
+                .await;
+            return;
+        }
+
+        let plan = match result {
+            Ok(ok) => ok,
+            Err(e) => return tx.send(Err(e), session),
+        };
+
+        self.sequence_plan(tx, session, Plan::CreateConnection(plan), depends_on)
+            .await;
+    }
```

Comment:
The `PeekValidity` struct does have some logic to handle things like this. If it could be reused and renamed to be more general, I'm on board with that.

Reply:
This is following the source purification pattern right above it. What is the issue that the other PR had?